tensorflow/java/BUILD

```diff
@@ -372,34 +373,48 @@ LINKER_VERSION_SCRIPT = ":config/version_script.lds"
 
 LINKER_EXPORTED_SYMBOLS = ":config/exported_symbols.lds"
 
-tf_cc_binary(
-    name = "libtensorflow_jni.so",
-    # Set linker options to strip out anything except the JNI
-    # symbols from the library. This reduces the size of the library
-    # considerably (~50% as of January 2017).
-    linkopts = select({
-        "//tensorflow:debug": [],  # Disable all custom linker options in debug mode
-        "//tensorflow:darwin": [
-            "-Wl,-exported_symbols_list",  # This line must be directly followed by LINKER_EXPORTED_SYMBOLS
-            "$(location {})".format(LINKER_EXPORTED_SYMBOLS),
-        ],
-        "//tensorflow:windows": [],
-        "//conditions:default": [
-            "-z defs",
-            "-s",
-            "-Wl,--version-script",  #  This line must be directly followed by LINKER_VERSION_SCRIPT
-            "$(location {})".format(LINKER_VERSION_SCRIPT),
+# System.loadLibrary() on OS X looks for ".dylib" or ".jnilib"
+# and on Windows looks for ".dll" instead of ".so".
+# If and when https://github.com/bazelbuild/bazel/issues/914
+# is resolved, perhaps this workaround rule can be removed.
+SHARED_LIBRARY_NAME_PATTERN = [
+    # prefix, extension
+    ("lib", ".so"),  # On Linux, shared libraries are usually named as libfoo.so
+    ("lib", ".dylib"),  # On macos, shared libraries are usually named as libfoo.dylib
+    ("", ".dll"),  # On Windows, shared libraries are usually named as foo.dll
+]
+
+[
+    tf_cc_binary(
+        name = prefix + "tensorflow_jni" + extension,
```

Comment:
Apologies for my bazel noobness here, but instead of generating all 3 targets, can a `select` be used on the name? Something like:

```
name = select({
  "//tensorflow:windows": "tensorflow_jni.dll",
  "//tensorflow:darwin": "libtensorflow_jni.dylib",
  "//conditions:default": "libtensorflow.so",
})
```

or something like that?

Reply:
Target name is not selectable, because you cannot have a target that only exists on a specific platform.